components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -439,6 +525,37 @@ public RuleTemplate getRuleTemplate(BusinessRuleFromTemplate businessRuleFromTem
 
     }
 
+    public List<RuleTemplate> getRuleTemplate(BusinessRuleFromScratch businessRuleFromScratch) throws
+            TemplateManagerException {
+        String templateGroupUUID = businessRuleFromScratch.getTemplateGroupUUID();
+        String[] ruleTemplateUUID = new String[2];
+        ruleTemplateUUID[0] = businessRuleFromScratch.getInputRuleTemplateUUID();
+        ruleTemplateUUID[1] = businessRuleFromScratch.getOutputRuleTemplateUUID();
+        TemplateGroup foundTemplateGroup = this.availableTemplateGroups.get(templateGroupUUID);
+        List<RuleTemplate> foundRuleTemplate = new ArrayList<>();
+
+
+        if (foundTemplateGroup != null) {
+            for (RuleTemplate ruleTemplate : foundTemplateGroup.getRuleTemplates()) {
+
+                for (String aRuleTemplateUUID : ruleTemplateUUID) {
+                    if (ruleTemplate.getUuid().equals(aRuleTemplateUUID)) {
+                        foundRuleTemplate.add(ruleTemplate);
+                    }
+                }
+            }
+            if (!foundRuleTemplate.isEmpty()) {
+                return foundRuleTemplate;
+            } else {
+                throw new TemplateManagerException("No rule template found with the given uuid");
```

Comment:
please add more context if can

Reply:
This was modified by 2c0119c4c48e1077512e685493252c388e68a520

Reply:
Fixed by b14dd18